compiler/rustc_ast/src/expand/autodiff_attrs.rs

```diff
@@ -6,27 +10,91 @@ use crate::expand::{Decodable, Encodable, HashStable_Generic};
 use crate::ptr::P;
 use crate::{Ty, TyKind};
 
-#[allow(dead_code)]
 #[derive(Clone, Copy, Eq, PartialEq, Encodable, Decodable, Debug, HashStable_Generic)]
 pub enum DiffMode {
+    /// No autodiff is applied (usually used during error handling).
     Inactive,
+    /// The primal function which we will differentiate.
     Source,
+    /// The target function, to be created using forward mode AD.
     Forward,
+    /// The target function, to be created using reverse mode AD.
     Reverse,
+    /// The target function, to be created using forward mode AD.
+    /// This target function will also be used as a source for higher order derivatives,
+    /// so compute it before all Forward/Reverse targets and optimize it through llvm.
     ForwardFirst,
```

Comment:
Remark: yes, I think the doc comment was needed, otherwise I wouldn't have guess what this meant, especially since the variant names does not match the enzyme docs (as they're meaningful to *us*).